WooCommerce/Classes/ViewRelated/Dashboard/Factories/WCAnalyticsStatsTotals+UI.swift

```diff
@@ -0,0 +1,91 @@
+import Yosemite
+
+/// Protocol for `WCAnalyticsStatsTotals` that can be parsed with the `StatsIntervalDataParser`
+protocol ParsableStatsTotals: WCAnalyticsStatsTotals {
+    /// Represents the types of stats total data
+    associatedtype TotalData: RawRepresentable
+
+    /// Gets the double value for the given stats total data type
+    func getDoubleValue(for data: TotalData) -> Double
+}
+
+// MARK: - Parsable Order Stats V4
+
+extension OrderStatsV4Totals: ParsableStatsTotals {
+    /// Represents a type of orders total data
+    enum TotalData: String {
+        case totalOrders
+        case totalItemsSold
+        case grossRevenue
+        case netRevenue
+        case averageOrderValue
+    }
+
+    func getDoubleValue(for data: TotalData) -> Double {
+        switch data {
+        case .totalOrders:
```

Comment:
Food for thought for other PR :) As mentioned in our call on Tuesday, there might be room to avoid duplicated lines e.g. 
```
case .totalOrders:
            return Double(totalOrders)
```
with some composition that gathers these properties in one class/protocol. That being said, there aren't many duplicated cases here, so we can leave it as it is.

Reply:
Thanks for mentioning that! I did think about trying to reduce that duplication here, but it seemed to overcomplicate things without adding much benefit. (As you mentioned, there aren't too many duplicated cases.) Since you're ok leaving it as it is for now, perhaps it's something to come back to if we continue adding more stats here and there's more overlap at that point.